openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -2166,6 +2166,88 @@ The following is a non-normative example of a Credential Response containing a C
+
+Since the key attestations may have large audience as many Credential Issuers that not necessarly uses the same trust framework or internal design decisions, it is required to use a common approach to facilitate interoperability. Therefore, key attestations SHOULD use a common format,allowing Issuers to develop consistent evaluation processes, reducing complexity and potential errors. Common formats makes easy for Issuers to demonstrate compliance with regulatory requirements across different jurisdictions, they also facilitate the development of shared best practices and security benchmarks.
+
+todo: explain usage of this within proof type or DPoP Proof
+
+## Key Attestation in JWT format
+
+The JWT is signed by the Wallet Provider or the Wallet's key storage component itself and contains the following elements:
+
+* in the JOSE header,
+  * `alg`: REQUIRED. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry [@IANA.JOSE.ALGS]. It MUST NOT be `none` or an identifier for a symmetric algorithm (MAC).
+  * `typ`: REQUIRED. MUST be `openid4vci-keyattestation+jwt`, which explicitly types the key proof JWT as recommended in Section 3.11 of [@!RFC8725].
+
+The key attestation may use `x5c`, `kid`, `trust_chain` or other mechanisms to convey the public key and the associated trust mechanism to sign the key attestation.
+
+* in the JWT body,
+  * `iat`: REQUIRED (number). Integer for the time at which the key attestation was issued using the syntax defined in [@!RFC7519].
+  * `exp`: REQUIRED (number). Integer for the time at which the key attestation expires using the syntax defined in [@!RFC7519].
+  * `keys` : REQUIRED. Array of attested keys using the syntax of JWK as defined in [@!RFC7517].
+  * `key_type` : OPTIONAL. String that asserts the key storage component and its security mechanism of attested keys from `keys`. This specification defines initial values in (#keyattestation-keytypes).
+  * `user_authentication` : OPTIONAL. String that asserts the security mechanism the key storage component uses to authenticate the End-User to authorize access to the private key from `keys`. This specification defines initial values in (#keyattestation-auth).
+  * `apr` : OPTIONAL. String that asserts the resistance to a certain attack potential as described
+  * `nonce`: OPTIONAL. String that represents a nonce provided by the Issuer to proof that a key attestation was freshly generated.
+  * `status`: OPTIONAL. JSON Object representing the supported revocation check mechanisms, such as the one defined in [status list]
+
```

Comment:
I suggest to also add an optional parameter that represents the key lifetime (like maximum duration for example). The Wallet solution provider might have reasons to enforce extra constraints and therefore he would need to specify to the Issuer to reduce the initially forseen validity of the EAA.

Reply:
My intention was to cover this with `exp`, do you think we need something additional or make the definition of `exp` more clearly?

Reply:
The current definition indicates that `exp` represents the expiration of the key attestation, which for me would be around an hour or so. I was thinking about something like `max_key_lifetime`, which is something that takes into account for example the key size and algorithm and estimates for how long the key will be considered as resistant (e.g. 3 years, 5 years, etc).

Reply:
What would be the benefit of the key attestation having a different expiration time? Where would this 1 hour be used?

Reply:
For me, having a short key attestation duration, like an hour indicates to the Issuer that he has a limited period of time within which he has to generate the EAA. The way I see it, it's more aligned with the request lifespan.
Which brings me to another question: do all the keys in the keys array have the same type? If not, it would make no sense to have one `max_key_lifetime` for all of them.

Reply:
My assumption was that all keys belong to the same WSCD/key storage component, I made a proposal to clarify that

Reply:
I still don't understand the usefulness of having 2 expiration times. Is there an attack scenario that you want to protect against? If the key is valid for 3-5 years, then this shouldn't change within one hour? If the Issuer wants replay protection then he should
a) provide a nonce that should we included to the key attestation
b) require a proof of possession with the key, like in credential request jwt proof type

Reply:
i think it is sufficient to clarify that exp claim applies to both key attestation and the keys they attesting - made the suggestion above.